### DISCUSSION TITLE
License error: Malformed: Failed to get license from the source: FromEnvVar{envName='MI_LICENSE'}

### DISCUSSION THREAD

**Discussion by jxshi** [milaboratory/mixcr#1413] [Q&A] [answered]
## Checklist before submitting the issue:

 - [x] The issue is strongly related to the MiXCR software
 - [x] The issue can be reproduced with [the most recent version](https://github.com/milaboratory/mixcr/releases) of MiXCR
 - [x] There is no answer to the question in the [official documentation](http://mixcr.readthedocs.io/) and there is no duplicate issue in the [bug tracker](https://github.com/milaboratory/mixcr/issues?utf8=✓&q=)  
 - [ ] Inspection of raw alignments with [exportAlignmentsPretty](http://mixcr.readthedocs.io/en/master/export.html#exporting-well-formatted-alignments-for-manual-inspection) shows that data has the expected architecture, and sample preparation artefacts are not the reason of the problem (if this is the matter of the issue)

## Expected Result

*A description of what you wanted to happen*
Server info:
```
Description:    Ubuntu 20.10
Release:        20.10
Codename:       groovy
```
Java Info:
```
openjdk 17.0.7 2023-04-18
OpenJDK Runtime Environment (build 17.0.7+7-Ubuntu-0ubuntu120.04)
OpenJDK 64-Bit Server VM (build 17.0.7+7-Ubuntu-0ubuntu120.04, mixed mode, sharing)
```
I run ```mixcr activate-license``` and entered the license, it showed 
```
Checking the license...
Ok
License activated successfully.
```
Then I run ```mixcr --help``` to check if it could work. Then error message popped up. I have tried to set ```MI_LICENSE``` variable in ```~/.bashrc``` file, put ```.mi.license``` file under ```$HOME``` directory. None of the tricks worked. 

By the way, I have also tried ```mixcr 4.4.1```. The issue could not be resolved by changing ```mixcr``` to a previous version.
## Actual Result
```
License manager thread died.
License error: Malformed: Failed to get license from the source: FromEnvVar{envName='MI_LICENSE'}
```

*A description of what actually happened*
## Exact MiXCR commands 
```mixcr --help```

## MiXCR report files 

*Paste here content of the report files produced by MiXCR (enclose it with triple ```` ``` ````  for better formatting)*

**Comment by PoslavskySV** [comment, marked as answer]
Hi,

the error message:
```
License error: Malformed: Failed to get license from the source: FromEnvVar{envName='MI_LICENSE'}
```
suggests that you have set `MI_LICENSE` env var with wrong content. `MI_LICENSE` var should be set with the content of you license code. If you use `MI_LICENSE_FILE` env var, then it should be set with the path to license file.

However, `activate-license` command should work without env vars at all. Could you please remove all these env vars, as well as mi.license file, and try to activate license once again? Before running `mixcr activate-license`, please execute:
```
mixcr deactivate-license
```

Let me know whether it works.

**Comment by jxshi** [comment]
Hi, 

Thank you for your suggestion. Your solution worked well. 

Best,
Jianxiang
